### PR TITLE
RFC: Use graphql-scalars

### PR DESCRIPTION
**Pull request by dkarnutsch**
As suggested in #2088, RFC for `graphql-scalars`.

TLDR: More explicit GraphQL schema.

Screenshot 2024-05-29 at 09 15 48

Useful scalars for us:
- Date
- EmailAddress
- Latitude
- Longitude
- Precise Numbers: (Positive) Float/Int
- URL
- UUID
- Hex Color Code

-----

Pro:
- It is easy to use
- Makes the schema more explicit and better usable
- Makes it easier to add "basic validation"

Con:
- Validation is split between class-validator and GraphQL schema (we have this situation already, but this would amplify it)

Proposed Migration-Path:
- Use [JSONObject](https://the-guild.dev/graphql/scalars/docs/scalars/json-object) from "graphql-scalars" (which is based on "graphql-type-json", what we already use)
- Rename our DateFilter to DateTimeFilter (as this is what it really is)
- Leave it to the application, if they want to use it
- Recreate #2088 by adding a DateFilter based on https://the-guild.dev/graphql/scalars/docs/scalars/date or allow the application to extend the filters

### CODE REVIEW

**Comment by nsams** [comment]
lgtm, i would say we should do that.

I'm not sure about changing `ID` though...

and another Con: You need to add it's type to codegen, this might be an argument against using PositiveInt

**Review by johnnyomair — COMMENTED** [review]
I'm not sure about this. I see the benefits for Date and DateTime, but the shown scalars UUID, PositiveInt and PositiveFloat don't offer much benefit IMO.
Also, would it even be possible to support this in the generator but not enforce it? How would this work for Date, where we would then have two different scalars?
We should discuss this in tomorrow's meeting.
